### PR TITLE
ARE-8166: Add direct Bearer token and Client Credentials

### PR DESCRIPTION
**Pull request by ajsteadly**
Support for supplying a token directly, or Client Credentials flow with refresh.

### CODE REVIEW

**Review by aravindg28 — APPROVED** [review]

**Comment by jc-verisk** [comment]
Have we considered using https://github.com/requests/requests-oauthlib which can handle token refreshing?

**Comment by ajsteadly** [comment]
> Have we considered using https://github.com/requests/requests-oauthlib which can handle token refreshing?

I have not, I will check it out

**Review by jc-verisk — APPROVED** [review]
lgtm

File: tests/test_requestor.py [hunk 1]
```diff
@@ -97,22 +98,150 @@ def test_malformed_response(self, reqmock):
...
+        assert reqmock.request_history[2].method == 'POST'
+        self._validate_token_params(reqmock.request_history[2].text)
+
+        assert reqmock.request_history[3].url == self.api_url
+        assert reqmock.request_history[3].method == 'GET'
+
+        # Expect to see the new token
+        assert reqmock.request_history[3].headers['Authorization'] == f'Bearer {second_mock_token}'
+
+    def test_retry_on_401(self, reqmock):
```
Comment (aravindg28) [inline tests/test_requestor.py:187]: Just curious, when will we get a 401 status code? Like what triggers it?

Comment (ajsteadly) [reply tests/test_requestor.py:187]: > needing to handle the case where an API call fails if an access token expires before the expected time. Access tokens can expire for many reasons, such as the user revoking an app, or if the authorization server expires all tokens when a user changes their password.

https://www.oauth.com/oauth2-servers/making-authenticated-requests/refreshing-an-access-token/

**Review by aravindg28 — APPROVED** [review]

**Review by DmitriiKh — APPROVED** [review]

**Review by jc-verisk — APPROVED** [review]

**Pull request merged**